### DISCUSSION TITLE
Modifying result of --collect-only

### DISCUSSION THREAD

**Discussion by Stefanhg** [pytest-dev/pytest#11846] [Q&A] [answered]
Hello,
While testing https://github.com/scastlara/pytest-tagging
I came across something I expected work but didn't.
In an nutshell the plugin allows you to mark an testcase with a tag and then `--tags (your_tag)` will only run those tests with the tag.

The tests are then selected/deselected in the `pytest_collection_modifyitems` hook.
When using `--tags (your_tag) --collect-only` it will still display all available tests and not just those within the tagged item.

I looked though the code and it seems like the handling of --collect-only is not done after the collection. Is there another way allowing me to modify the collected result for `--collect-only` or will I have to add another custom argument to display the tagged items?

**Comment by bluetech** [comment, marked as answer]
>  When using --tags (your_tag) --collect-only it will still display all available tests and not just those within the tagged item.

Hmm, it *should* work. I just tested it with pytest's built-in `-k` and `-m` filters, which are both implemented with `pytest_collection_modifyitems` internally, and `--collect-only` only shows the filtered ones and the summary line reads `34/3551 tests collected (3517 deselected)`.

One thing that I'd try is change pytest-tagging `pytest_collection_modifyitems` implementation to not be a hookwrapper, I don't think there is any reason for it to be. I don't think it's related to what you're seeing but worth a try anyway.

**Reply by Stefanhg** [reply]
@bluetech Thanks for your reply.
I tried to remove the `hookwrapper=True` but this did not solve anything
If i add some debug info to `pytest_collection_modifyitems` it never enters the function when collecting. I also tried `@pytest.mark.tryfirst`

After some debugging it turns out for some reason this exact feature was actually bypassed! In `pytest_configure` it had a `if not config.option.collectonly:` in the register of the TaggerRunner class!
